tests/PageObjects/Locator.py

```diff
@@ -0,0 +1,349 @@
+                    else:
+                        print("Unkown requirement in locator")
+                return False
+
+        if self.__VERBOSE__:
+            print("Ready to perform action on {} again...".format(self.description))
+        return True
+
+    # Will print element found status using provided description
+    def __describe__(self, descr, found):
+        if not self.__VERBOSE__:
+            return
+        msg = "READY!"
+        if not found:
+            msg = "NOT READY!"
+        if descr != "":
+            print("[{}] {}".format(descr, msg))
+        else:
+            print("[Element] {}".format(msg))
+
+    def __search_element__(self):
+        # Find the element using specified method
+        if self.__TRY__:
+            self.__lazy_search__()
+        else:
+            self.__hard_search__()
+
+        # Perform validation of state
+        valid_state = self.__valid_state__()
+
+        # Reset modifiers to defaults after validation
+        self.__reset_defaults__()
+
+        return valid_state
+
+    # Will find the locator element based on the modifier selected
+    # extra_args -> any extra arguments to pass to action besides element
+    def __perform__(self, action, descr, *extra_args):
+        # Perform the specified action if element found
+        if self.__search_element__():
+            new_args = (self.element,) + extra_args
+            return Action(action, descr, *new_args).perform(self.__VERBOSE__)
+
+    # Will find the element using the find actions, returns element(s) found
+    def __find_element__(self, locator, locator_type):
+        # Valid locator types
+        valid = ["id", "class", "css", "xpath"]
+        if locator_type not in valid:
+            raise ValueError("Invalid locator type: {}".format(locator_type))
+        try:
+            # Get the proper locator method
+            method = self.locator_type_to_method(locator_type)
+
+            # If its a child element, find element within current element
+            if self.is_child:
+                if self.element is None:
+                    return None
+                if self.is_multiple:
+                    return self.element.find_elements(method, locator)
+                return self.element.find_element(method, locator)
+
+            # Not a child element, return element using driver
+            if self.is_multiple:
+                return self.driver.find_elements(method, locator)
+            return self.driver.find_element(method, locator)
+        except NoSuchElementException:
+            return None
+
+    # Will try to find element and print status, will quit if not found
+    # Does not raise error if element was not found.
+    def __lazy_search__(self):
+        if self.locator is None:
+            self.element = None
+            return
+        # Attempt to find element
+        self.element = self.__find_element__(self.locator, self.locator_type)
+
```

Comment:
do we need to check if self.element is None before we move on to the next if statement?